app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/startregistration/handlers/StartRegistrationHandler.kt

```diff
@@ -0,0 +1,87 @@
+package com.x8bit.bitwarden.ui.auth.feature.startregistration.handlers
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import com.x8bit.bitwarden.data.platform.repository.model.Environment
+import com.x8bit.bitwarden.ui.auth.feature.startregistration.StartRegistrationAction
+import com.x8bit.bitwarden.ui.auth.feature.startregistration.StartRegistrationViewModel
+
+/**
+ * A collection of handler functions for managing actions within the context of the
+ * [StartRegistrationScreen].
+ */
+data class StartRegistrationHandler(
+    val onEmailInputChange: (String) -> Unit,
+    val onNameInputChange: (String) -> Unit,
+    val onEnvironmentTypeSelect: (Environment.Type) -> Unit,
+    val onContinueClick: () -> Unit,
+    val onTermsClick: () -> Unit,
+    val onPrivacyPolicyClick: () -> Unit,
+    val onReceiveMarketingEmailsToggle: (Boolean) -> Unit,
+    val onUnsubscribeMarketingEmailsClick: () -> Unit,
+    val onServerGeologyHelpClick: () -> Unit,
+    val onBackClick: () -> Unit,
+) {
+
+    companion object {
+        /**
+         * Creates an instance of [StartRegistrationHandler] by binding actions to the provided
+         * [StartRegistrationViewModel].
+         */
+        fun create(viewModel: StartRegistrationViewModel): StartRegistrationHandler {
+            return StartRegistrationHandler(
+                onEmailInputChange = {
+                    viewModel.trySendAction(
+                        StartRegistrationAction.EmailInputChange(
+                            it,
+                        ),
+                    )
+                },
+                onNameInputChange = {
+                    viewModel.trySendAction(
+                        StartRegistrationAction.NameInputChange(
+                            it,
+                        ),
+                    )
+                },
+                onEnvironmentTypeSelect = {
+                    viewModel.trySendAction(
+                        StartRegistrationAction.EnvironmentTypeSelect(
+                            it,
+                        ),
+                    )
+                },
+                onContinueClick = {
+                    viewModel.trySendAction(StartRegistrationAction.ContinueClick)
+                },
+                onTermsClick = { viewModel.trySendAction(StartRegistrationAction.TermsClick) },
+                onPrivacyPolicyClick = {
+                    viewModel.trySendAction(StartRegistrationAction.PrivacyPolicyClick)
```

Comment:
Do we not have tests for all these clicks?

Reply:
Added some additional tests to cover the remaining actions, sans the ones in the annotated strings. I cannot for the life of me find reliable way to get them to invoke even with the CustomAccessibilityActions

Reply:
Gotcha, we have tried to stay away from those since they aren't great for testing or accessibility but given the design, I am not sure how else to do it.

Reply:
I actually got it, I will add.